3b.run_sam/scripts/1.run_sam2_microscopy.py

```diff
@@ -0,0 +1,615 @@
+    # convert the labels into position coordinates
+    regions = regionprops(label(labels))
+    coords = np.array([r.centroid for r in regions])
+
+    # plot the points and the masks and the image side by side by side
+    fig, ax = plt.subplots(1, 4, figsize=(30, 15))
+    ax[0].imshow(img, cmap="gray")
+    ax[0].set_title("Image")
+    ax[1].imshow(render_label(labels, img=img))
+    ax[1].set_title("Masks")
+    ax[2].imshow(img, cmap="gray")
+    ax[2].scatter(
+        coords[:, 1],
+        coords[:, 0],
+        color="red",
+        marker="*",
+        s=100,
+        edgecolor="white",
+        linewidth=1.25,
+    )
+    ax[2].set_title("Points")
+    coords = coords[:, [1, 0]]
+    ax[3].invert_yaxis()
+    # make the aspect ratio equal
+    ax[3].set_aspect("equal")
+    show_points(coords, np.ones(len(coords)), ax[3])
+    labels = np.ones(coords.shape[0], dtype=np.int32)
+    image_set_dict["image_x_y_coords"].append(coords)
+    image_set_dict["image_labels"].append(labels)
+
+# remove star dist model from memory
+del model
+# remove all stardist gpu memory
+torch.cuda.empty_cache()
+
+
+# ## 4. Track multiple objects in the video
+
+# In[ ]:
+
+
+image_set_dict.keys()
+
+
+# ### Begin GPU Profiling
+
+# In[ ]:
+
+
+# Start recording memory snapshot history
+logging.basicConfig(
+    format="%(levelname)s:%(asctime)s %(message)s",
+    level=logging.INFO,
+    datefmt="%Y-%m-%d %H:%M:%S",
+)
+logger: logging.Logger = logging.getLogger(__name__)
+logger.setLevel(level=logging.INFO)
+
+TIME_FORMAT_STR: str = "%b_%d_%H_%M_%S"
+# delete any prior memory profiling data
+delete_recorded_memory_history(
+    logger=logger, save_dir=pathlib.Path("../memory_snapshots/").resolve()
+)
+
+# Keep a max of 100,000 alloc/free events in the recorded history
+# leading up to the snapshot.
+MAX_NUM_OF_MEM_EVENTS_PER_SNAPSHOT: int = 100000
+start_record_memory_history(
+    logger=logger, max_entries=MAX_NUM_OF_MEM_EVENTS_PER_SNAPSHOT
+)
+
+
+# In[ ]:
+
+
+# clear the memory
+torch.cuda.empty_cache()
+gc.collect()
+
+
+# In[ ]:
+
+
+stored_video_segments = {}
+
+
+# In[ ]:
+
+
+# loop through each image set and predict the instances
+for i in range(len(image_set_dict["image_set_name"])):
+    print(
+        f"{image_set_dict['image_set_name'][i]}: {image_set_dict['image_set_first_frame'][i]}"
+    )
+    frame_names = sorted(list(Path(image_set_dict["image_set_path"][i]).glob("*.jpeg")))
+    img = io.imread(frame_names[0])
+    h, w = img.shape
+    print(h, w)
+    # initialize the state
+    inference_state = predictor.init_state(
+        video_path=str(image_set_dict["image_set_path"][i]),
+        offload_video_to_cpu=True,
+        offload_state_to_cpu=True,
+    )
+    predictor.reset_state(inference_state)
+    prompts = {}
+    ann_frame_idx = 0
+    ann_obj_idx = 1
+    samples = 1
+    negative_sampling = (
+        False  # set True to generate negative samples for better training
```

Comment:
Recommend removing this code comment to avoid confusion for why you don't have it set to True.

Reply:
Haha, again not enough GPU RAM so it is set to false